### PR TITLE
Modified platform utils based on platform type and not SKU

### PR DESCRIPTION
**Pull request by madhanmellanox**
…onaries and functions in script files sfputil.py, chassis.py, fan.py, psu.py and thermal.py

**- What I did**
I moved all the SKU based variables (dictionaries), functions to Platform based variables and functions in scripts like sfputil.py, chassis.py, psu.py, fan.py and thermal.py

**- How I did it**
I eliminated almost all the references of SKU based dictionaries and functions from these files and instead used Platform based dictionaries and functions.

**- How to verify it**
Used the following commands to verify my changes are working:
sudo sfputil show presence
show platform psustatus
show interface transceiver eeprom Ethernet0
show platform temperature

**- Description for the changelog**

The changes are only in files sfputil.py, chassis.py, fan.py, psu.py and thermal.py and the changes are conversion of dictionaries and functions from SKU based to Platform based.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 1]
```diff
@@ -33,15 +33,15 @@
...
 
 # Ethernet<n> <=> sfp<n+SFP_PORT_NAME_OFFSET>
 SFP_PORT_NAME_OFFSET = 1
 SFP_PORT_NAME_CONVENTION = "sfp{}"
 
-# magic code defnition for port number, qsfp port position of each hwsku
+# magic code defnition for port number, qsfp port position of each platform
 # port_position_tuple = (PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET)
-hwsku_dict = {'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, "LS-SN2700":0, 'ACS-MSN2740': 0, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 3, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4}
+platform_dict = {'x86_64-mlnx_msn2700-r0': 0, 'x86_64-mlnx_msn2740-r0': 0, 'x86_64-mlnx_msn2100-r0': 1, 'x86_64-mlnx_msn2410-r0': 2, 'x86_64-mlnx_msn2010-r0': 3, 'x86_64-mlnx_msn3700-r0': 0, 'x86_64-mlnx_msn3700c-r0': 0, 'x86_64-mlnx_msn3800-r0': 4}
```
Comment (keboliu) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: missing  x86_64-mlnx_msn4700-r0  platform

**Review by Junchao-Mellanox — APPROVED** [review]

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py (hunk 1 above)
Comment (madhanmellanox) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: Addressed in the next commit. Thanks.

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
 MLNX_NUM_PSU = 2
 
 GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:31]: is this still needed and used?

**Review by liat-grozovik — COMMENTED** [review]
@madhanmellanox  can you please address my question if the SKU define is still needed?
also, can we add sn3420 and sn4600c to the list even if there are not fully supported?  there are PRs in progress and not sure what will be merged first.

**Comment by liat-grozovik** [comment]
suggest also the rephrase the commit header.
it is too long. i think the idea is to have platform utils based on platform type and not SKU.

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py (hunk 2 above)
Comment (madhanmellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:31]: This is still required to set the name of the chassis, as it is done in the below code:
        mi = get_machine_info()
        if mi is not None:
            self.name = mi['onie_platform']
        else:
            self.name = self.sku_name

**Comment by madhanmellanox** [comment]
> @madhanmellanox can you please address my question if the SKU define is still needed?
> also, can we add sn3420 and sn4600c to the list even if there are not fully supported? there are PRs in progress and not sure what will be merged first.

SKU defines is required in chassis.py, just to name the chassis, as mentioned in the above comment. Will add SN3420 and SN4600c in the next commit.

**Review by liat-grozovik — CHANGES_REQUESTED** [review]
please create new PR as yours has 1006 files changed.
we need to be sure to review only relevant files.

and please make sure to test such before you are asking for review.

**Comment by madhanmellanox** [comment]
Since, the PR is polluted with 2 many files, I am closing this PR.

**Pull request closed without merging**